components/profile/pages/wallet/wallet-balance/controller.tsx

```diff
@@ -0,0 +1,194 @@
+  }
+
+  function handleSearchFilter(name = "", symbol = "", networks) {
+    const hasNetworkName = query?.networkName
+    const isNetwork = !!networks.find(({ name }) =>
+      hasNetworkName?.toString().toLowerCase() === name?.toLowerCase());
+
+    return (
+      (name.toLowerCase().indexOf(search.toLowerCase()) >= 0 ||
+        symbol.toLowerCase().indexOf(search.toLowerCase()) >= 0) &&
+      (hasNetworkName ? isNetwork : true)
+    );
+  }
+
+  function loadBalances() {
+    if (!state.currentUser?.walletAddress) return;
+
+    searchCurators({
+      address: state.currentUser?.walletAddress,
+      chainShortName:
+        query?.chain?.toString() || state?.connectedChain?.shortName,
+    }).then(({ rows }) => {
+      Promise.all(rows?.map(async (curator) => {
+        const tokenInformation = await getCoinInfoByContract(curator?.network?.networkToken?.symbol);
+
+        return {
+            symbol: t("$oracles", {
+              token: curator?.network?.networkToken?.symbol,
+            }),
+            name: `${t("misc.locked")} ${curator?.network?.networkToken?.name}`,
+            address: curator?.network?.networkToken?.address,
+            icon: <TokenIcon src={tokenInformation?.icon as string} />,
+            oraclesLocked: BigNumber(curator.tokensLocked),
+            networkName: curator?.network?.name,
+        };
+      })).then(setTokensOracles);
+    });
+
+    if (state.Service?.starting) return;
+
+    getTokens(state?.connectedChain?.id).then((tokens) => {
+      Promise.all(tokens?.map(async (token) => {
+        const tokenData = await processToken(token?.address);
+        return { networks: token?.networks, ...tokenData };
+      })).then(setTokens);
+    });
+  }
+
+  useEffect(loadBalances, [
+    state.currentUser?.walletAddress,
+    state.connectedChain,
+    state.Service?.starting,
+  ]);
+
+  useEffect(() => {
+    if (!tokens?.length) return;
+
+    Promise.all(tokens.map(async (token) => ({
+        tokenAddress: token.address,
+        value:
+          typeof token.balance === "string"
+            ? BigNumber(token.balance)
+            : token.balance,
+        price: await getCoinPrice(token?.symbol,
+                                  state?.Settings.currency.defaultFiat),
+    }))).then((tokens) => {
+      const totalConverted = tokens.reduce((acc, token) =>
+          BigNumber(token.value)
+            .multipliedBy(token.price || 0)
+            .plus(acc),
+                                           BigNumber(0));
+      const noConverted = !!tokens.find((token) => token.price === undefined);
+      const totalTokens = tokens.reduce((acc, token) => BigNumber(token.value).plus(acc),
+                                        BigNumber(0));
+
+      setTotalAmount(noConverted ? totalTokens.toFixed() : totalConverted.toFixed());
+      setHasNoConvertedToken(noConverted);
+    });
+  }, [tokens]);
+
+  useEffect(() => {
+    if(!query?.networkName && query?.network){
```

Comment:
shouldn't we just _not_ come to this page with `query.network` and instead have `query.networkName` be the default?